pkg/framework/aries/api/protocol.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package api
+
+import (
+	"github.com/go-openapi/runtime/middleware/denco"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+)
+
+//Handler http handler for each controller API endpoint
+type Handler interface {
+	Path() string
+	Method() string
+	Handle() denco.HandlerFunc
+}
+
+// ProtocolSvc interface for protocol service
+type ProtocolSvc interface {
+	GetAPIHandlers() []Handler
+}
+
+// Provider interface for protocol ctx
+type Provider interface {
+	OutboundTransport() transport.OutboundTransport
+}
+
+// NewProtocolSvc method to create new protocol service
+type NewProtocolSvc func(prv Provider) (ProtocolSvc, error)
```

Comment:
Suggest we rename. I find it confusing for a type to be prefixed with New.

Reply:
I will change it to ProtocolSvcCreator

Reply:
done